### PR TITLE
Update UG, DG, EditCommand and EditCommandTest

### PR DESCRIPTION
**Pull request by pranav-ganesh**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -355,7 +355,7 @@ This feature allows telemarketers to edit data fields at any point in time if th
 
 Given below is an example usage scenario and how the edit mechanism behaves at each step.
 
-Step 1. The user launches the application and realizes that person 1 needs to be edited. He/she executes the command `edit 1 name/Will age/20` in order to edit the first person in the address book.
+Step 1. The user launches the application and realizes that person 1 needs to be edited. He/she executes the command `edit 1 n/Will age/20` in order to edit the first person's name and age in the address book.
```
Comment (loose-bus-change) [inline docs/DeveloperGuide.md:358]: use contacts. 
and also 'realizes that the first contact' or 'realizes that the details of the first contact need to be'

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -386,29 +394,29 @@ The following activity diagram summarizes what happens when a user executes an e
...
 ### Design considerations:
 
-**Aspect: Compulsory fields:**
+**Aspect: Duplicate detection:**
 
-* **Alternative 1 (current choice):** 3 compulsory fields (`Name`, `Email` and `Phone`)
+* **Alternative 1 (current choice):** 3 fields (`Name`, `Email` and `Phone`) need to exactly match for duplicate to be detected
 
-    * Pros: Enhance user experience since they don't need to waste time entering many data field values
+    * Pros: More logical and emulates real world scenario, It is impossible for non-duplicate contacts to have name, email and phone
```
Comment (loose-bus-change) [inline docs/DeveloperGuide.md:401]: 'to have same name'

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -963,15 +1017,40 @@ testers are expected to do more *exploratory* testing.
...
+### Editing a person
+
+1. Editing a contact while all contacts are being shown
+
+2. Editing the name of the contact
+  * Test case: `edit 1 n/bob`
+    Expected: The name of the first contact is edited to 'bob'
+
+3. Editing the age and interests of the contact
+  * Test case `edit 2 age/22 i/(1) shopping`
```
Comment (houtenteo) [inline docs/DeveloperGuide.md:1038]: Editing multiple fields at once (name and age)
Dont test interest here

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -963,15 +1017,40 @@ testers are expected to do more *exploratory* testing.
...
+    Expected: The age of the second contact as well as the first item in its interests list are edited
+
+4. Editing the interests list of a specific contact
+  * Test case `edit 4 i/(1) remove i/run`
```
Comment (houtenteo) [inline docs/DeveloperGuide.md:1042]: I think you can split up this test case into 3. So one to add and interest, another to edit an interest and another one to remove. Then u can have one more test case that does all 3 at once.

File: docs/UserGuide.md [hunk 5]
```diff
@@ -190,9 +186,11 @@ Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [g/GENDER] [age/AGE
 </div>
 
 Examples:
-* `edit 1 n/Malan i/[2] Swimming` edits the name of contact and the second interest of the first contact in the displayed list
-* `edit 1 i/Painting i/[1] Running` adds 'painting' as an interest and edits the first interest of the first contact in the displayed list
+* `edit 1 n/Malan i/(2) Swimming` edits the name of contact and the second interest of the first contact in the displayed list
```
Comment (houtenteo) [inline docs/UserGuide.md:189]: change to "edits the name of the first contact in the displayed list to 'malan' and its second interest to 'swimming'

File: docs/UserGuide.md [hunk 6]
```diff
@@ -190,9 +186,11 @@ Format: `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [g/GENDER] [age/AGE
...
+* `edit 1 i/Painting i/(1) Running` adds 'painting' as an interest and edits the first interest of the first contact in the displayed list
 * `edit 2 g/M e/myEmail@email.com age/55` edits the gender, email and age of the second contact in the displayed list
+* `edit 3 i/golf i/(1) eat i/(2) remove` removes the second interest in the interests list, edits the first item of the interests list to 'eat'
```
Comment (houtenteo) [inline docs/UserGuide.md:192]: "edits the first item...." should be "edits the first interest"

Comment (houtenteo) [reply docs/UserGuide.md:192]: Dont have to say "interests list" everytime. Just the first time will do. for eg. the second one can just be "edits the first interest to 'eat'"

File: docs/UserGuide.md [hunk 7]
```diff
@@ -508,7 +495,7 @@ Action | Format, Examples
 **Add** | `add n/NAME p/PHONE_NUMBER e/EMAIL [a/ADDRESS] [g/GENDER] [age/AGE] [i/INTEREST]…​` <br> e.g. `add n/Labuschagne Ho p/22224444 e/labuschagne@example.com a/my house g/F age/95 i/Sleeping`
 **List** | `list`
 **Called** | `called INDEX` <br> e.g., `called 2`
-**Edit** | `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [g/GENDER] [age/AGE] [c/CALLED] [i/[INTERESTSLIST INDEX] INTEREST]` <br> e.g., `edit 1 n/Bob p/68889444 e/email@email.com a/his house  g/M age/33 i/Eating i/[2] Swimming`
+**Edit** | `edit INDEX [n/NAME] [p/PHONE] [e/EMAIL] [a/ADDRESS] [g/GENDER] [age/AGE] [c/CALLED] [i/(OPTIONAL INTERESTSLIST INDEX) INTEREST]... [i/(INTERESTSLIST INDEX) remove]... [i/INTEREST]..` <br/> e.g., `edit 1 n/Bob p/68889444 e/email@email.com a/his house g/M age/33 i/Eating i/(2) Swimming`
 **Delete** | `delete INDEX`<br> e.g., `delete 3`
```
Comment (houtenteo) [inline docs/UserGuide.md:499]: <br/> should be <br>

File: docs/diagrams/EditActivityDiagram.puml [hunk 8]
```diff
@@ -1,16 +1,28 @@
...
+    if() then ([Throws Command Exception])
+      :Outputs error message;
+      else ([else])
+      :Returned person object is compared with the person to edit;
+      if() then ([the two persons are not equal and the model contains the edited person])
+        :Outputs error message;
+        else ([two persons are equal])
+        :Model sets the person to edit to the edited person;
+        :Edited person is displayed on the user interface;
 endif
```
Comment (houtenteo) [inline docs/diagrams/EditActivityDiagram.puml:24]: add an [else] before the endif

File: docs/diagrams/EditActivityDiagram.puml [hunk 9]
```diff
@@ -1,16 +1,28 @@
...
-    :EditCommand replaces the person to be edited
-    with a new person object containing the updated
-    data fields;
-else ([else])
+  :All the arguments are initialized in an EditPersonDescriptor object;
+  :New EditCommand object is created;
+  if() then ([invalid index])
+    :Outputs error message;
+    else ([else])
+    :CreateEditedPerson method in EditCommand creates and returns a new person object containing the updated data fields;
```
Comment (houtenteo) [inline docs/diagrams/EditActivityDiagram.puml:14]: "CreateEditedPerson method in EditCommand creates and returns a new person object containing the updated data fields;" change to "EditCommand creates and returns a new person object \n containing the updated data fields;"

(remove createEditedPerson, add newline after person object)

File: docs/diagrams/EditActivityDiagram.puml [hunk 10]
```diff
@@ -1,16 +1,28 @@
...
+    if() then ([Throws Command Exception])
```
Comment (houtenteo) [inline docs/diagrams/EditActivityDiagram.puml:15]: change this to the else block.

Old else block will become the new if (Contact is valid)

File: docs/diagrams/EditActivityDiagram.puml [hunk 11]
```diff
@@ -1,16 +1,28 @@
...
+      :Returned person object is compared with the person to edit;
+      if() then ([the two persons are not equal and the model contains the edited person])
+        :Outputs error message;
+        else ([two persons are equal])
+        :Model sets the person to edit to the edited person;
+        :Edited person is displayed on the user interface;
 endif
+
 stop
+
```
Comment (houtenteo) [inline docs/diagrams/EditActivityDiagram.puml:27]: Remove the last if else and remove "Returned person object is compared with the person to edit;"

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 12]
```diff
@@ -41,27 +42,29 @@
...
             + "[" + PREFIX_PHONE + "PHONE] "
             + "[" + PREFIX_EMAIL + "EMAIL] "
             + "[" + PREFIX_ADDRESS + "ADDRESS] "
             + "[" + PREFIX_CALLED + "CALLED]"
             + "[" + PREFIX_GENDER + "GENDER] "
             + "[" + PREFIX_AGE + "AGE] "
-            + "[" + PREFIX_INTEREST + "(optional index) INTEREST]...\n"
+            + "[" + PREFIX_INTEREST + "INTEREST_TO_BE_ADDED]"
+            + "[" + PREFIX_INTEREST + "(INDEX) remove]"
+            + "[" + PREFIX_INTEREST + "(OPTIONAL INDEX) INTEREST]...\n"
```
Comment (houtenteo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:55]: change "..." to the ones we use in the UG
remove "optional" from "optional index"

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 13]
```diff
@@ -41,27 +42,29 @@
...
             + "Example: " + COMMAND_WORD + " 1 "
             + PREFIX_PHONE + "91234567 "
             + PREFIX_EMAIL + "johndoe@example.com "
             + PREFIX_GENDER + "M "
             + PREFIX_CALLED + "false"
-            + PREFIX_INTEREST + "[1] software engineering";
+            + PREFIX_INTEREST + "[1] software engineering"
+            + PREFIX_INTEREST + "[2] remove";
```
Comment (houtenteo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:62]: Should be in round brackets

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 14]
```diff
@@ -153,19 +156,25 @@ private Person createEditedPerson(Person personToEdit, EditPersonDescriptor edit
      * Edits the {@code InterestsList} attribute of {@code personToEdit} based on user input command.
      */
     public void editInterestList(InterestsList newList, InterestsList currentList) throws CommandException {
+        this.emptyLists();
+
```
Comment (houtenteo) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:160]: remove 'this.' for class method calls

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 15]
```diff
@@ -21,6 +21,9 @@
  */
 public class EditCommandParser implements Parser<EditCommand> {
 
+    // public static String MESSAGE_INDEX_NOT_PARSED = "The index entered is not within the allowed range. "
+    //        + "The index cannot be bigger than 2147483647 (i.e., MAX_VALUE).";
+
```
Comment (houtenteo) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:26]: delete this comment

**Review by houtenteo — APPROVED** [review]
LGTM

**Pull request merged**